### DISCUSSION TITLE
Request about correct modelling of SiO2

### DISCUSSION THREAD

**Discussion by QifengHu** [NanoComp/meep#2909] [Q&A]
We are testing the performance of MEEP, in a grating structure with a long-wavelength range. The results of the pre-defined SiO2 are not desirable, since the valid wavelength range is limited from 0.25 to 1.77. We define a medium with fixed refractive index 1.46 and 1.55, the results is also not good in the long range, as followed: 
![silica_effect_of_refraidx1 46_R_only_SourceH8_TranFluxH6_ReflFluxH6_200px](https://github.com/user-attachments/assets/637a611e-3ec2-46ce-8aff-730c94afefad)
![silica_effect_of_refraidx1 55_R_only_SourceH8_TranFluxH6_ReflFluxH6_200px](https://github.com/user-attachments/assets/2db020b3-c163-4dfa-afaa-32b510f5cc2b)
We also tested the model of fused quartz, the transmittance in long wavelength range still impacts on the monitored region of transmittance.
![silica_effect_of_default_fused_quartz_R_T_SourceH8_TranFluxH2_ReflFluxH6_200px](https://github.com/user-attachments/assets/3be5b65f-a856-4555-8afe-0f43eda42c58)
![silica_effect_of_default_fused_quartz_R_T_SourceH8_TranFluxH4_ReflFluxH6_200px](https://github.com/user-attachments/assets/e980b406-25d5-45bc-9ce6-46443432a57a)
![silica_effect_of_default_fused_quartz_R_T_SourceH8_TranFluxH6_ReflFluxH6_200px](https://github.com/user-attachments/assets/f132872c-b01b-4f87-a65f-0397d5c15ffa)

Is it possible to implement the correct model of SiO2 with the indices in the long wavelength range?

**Comment by mdfeinstein** [comment]
It is possible. Materials in meep need to be fitted as a sum of Lorentzians (or Drude) susceptibilities. [In the docs](https://meep.readthedocs.io/en/latest/Materials/#how-do-i-import-n-and-k-values-into-meep), they supply a script that can kind of do this, but it isn't the easiest to use. I wrote my own function based on that approach and incorporating much of that code, but its lacking some features I wanted to put in. I can try to share it soon as is if that might be helpful, but I use the nlopt package the script uses, which needs to be [downloaded and built](nlopt.readthedocs.io), so its some effort to get up and running. (Side note: I think scipy has fitting functions that would be sufficient and much easier to just conda install into your environment, but I didn't test that.)

In the meantime, below is the code to define silica in the near/mid-infrared. I fitted from the [Kischkat data](https://refractiveindex.info/?shelf=main&book=SiO2&page=Kischkat). There is good agreement for the range you are interested in, but I will note that the absorption bands peaking at 9.3 um are too simplistic in this fit. Its really a few Lorentzians, but this fit sees it as only one. 
`SiO2 = mp.Medium( epsilon = 1.1,
               E_susceptibilities = [ 
                   mp.LorentzianSusceptibility(frequency=3.8146083476490666,
                                          gamma=4.301154772189638,
                                          sigma=9.422139872583635e-181),
                mp.LorentzianSusceptibility(
                    frequency=0.10525565241548465,
                    gamma=0.006728273595379233,
                    sigma=0.6703341698663918,
                ),
                mp.LorentzianSusceptibility(
                    frequency=1.53822387094153,
                    gamma=0.0,
                    sigma=0.9250921216802024,
                )  
               ]
              )
`
![SiO2 Kischkat fit1](https://github.com/user-attachments/assets/44425bb5-e790-4c0a-b6bb-4d39455df21a)
